### DISCUSSION TITLE
Toggle Fan SVG based on state of another entity

### DISCUSSION THREAD

**Discussion by MadMirza** [ExperienceLovelace/ha-floorplan#317] [Q&A] [answered]
<img width="150" alt="image" src="https://user-images.githubusercontent.com/100464787/211273517-a730c268-92c6-42ef-aa1b-d387a8b132e3.png">

hi, I have a fan that not smart enough to reflect the state correctly. But I have a wall switch that has energy reading, was trying to make the fan spin if the wattage reading is above 40 watts. I am quite bad at coding, takes me ages to figure out stuff even using tutorials, any help would be wonderful.
        - name: Fans
          entities:
            - entity: fan.nikheeil_room_fan
              element: fan
          tap_action:
            action: toggle
          hold_action: more-info
          state_action:
            service: floorplan.class_set
            service_data: >-
              '${(sensor.wall_switch_nikheeil_2_power.state >== 40) ? "show":"spinning"}'

**Comment by OzGav** [comment, marked as answer]
There is a big series of posts that @exetico did to show a variety of ways to do this (https://github.com/ExperienceLovelace/ha-floorplan/discussions/264#discussioncomment-4171701) but this should work:
```
- name: Fans
  entity: fan.nikheeil_room_fan
  element: fan
  tap_action: toggle
  hold_action: more-info
  state_action:
    service: floorplan.class_set
    service_data: |
      >
        if(hass.states["sensor.wall_switch_nikheeil_2_power"].state >= 40)
          return "spinning";
        else
          return "show"; 
```

P.S. Next time to make it easier to help you please format your code by highlighting it and pressing the <> button.

**Reply by MadMirza** [reply]
That worked like a charm! 
Thanks for the extra tip.

**Reply by MadMirza** [reply]
@OzGav ohh it actually works, but i gotta refresh everytime to start and stop the spinning. Which means even if i switch off the fan and the wattage goes below 40 it spins until I refresh the page. 
I am about to ask a very noob question, what language is the below part, is that Javascript. I am totally just mixing and matching things that i see in forums and elsewhere.
`      >
        if(hass.states["sensor.wall_switch_nikheeil_2_power"].state >= 40)
          return "spinning";
        else
          return "show"; `

**Reply by OzGav** [reply]
Yes it is JavaScript. The floorplan will update when fan.nikheeil_room_fan changes state. If the fan entity is that slow to update then you might need to use some automations and maybe an input_boolean to control the fan and keep track of its state.

**Reply by OzGav** [reply]
You could use a script as a tap action to control the fan.
